OpenRA.Game/Player.cs

```diff
@@ -196,7 +196,13 @@ public override string ToString()
 		public Dictionary<Player, Stance> Stances = new Dictionary<Player, Stance>();
 		public bool IsAlliedWith(Player p)
 		{
-			// Observers are considered allies to active combatants
+			if (p != null && p.Spectating && !NonCombatant && p.World.RenderPlayer != null)
+			{
```

Comment:
OpenRA code style is to omit braces from single line blocks. Please remove these and move the comment above the `if()`.